### PR TITLE
Fix #5926 Show toolitps for all disabled buttons (Chrome)

### LINKED ISSUE #5926: Show toolitps for all disabled buttons (Chrome)
## Description

Following the discussion [here](https://github.com/geosolutions-it/MapStore2/pull/5872#pullrequestreview-490593416) we would like to apply the same fix or use a general component (maybe editing the already used one, e.g. OverLayTrigger)

We should fix all the buttons that have this problem

## How to reproduce

-  find a disabled button, like locate button for testing

*Expected Result*

image

*Current Result*

image
on the left the is the problem when the button is disabled

- [x] browser related Chrome and Chromium, Edge

<b>Browser info</b> 

(use this site: <a href="https://www.whatsmybrowser.org/">https://www.whatsmybrowser.org/</a> for non expert users)

| Browser Affected | Version |
|---|---|
|Internet Explorer| not supported anymore from us |
|Edge| 85.0.564.51  |
|Chrome| 85.0.4183.121  |
|Firefox| 80.0.1 (64-bit), it works |
|Safari| not sure |

### PR DESCRIPTION
**Pull request by vlt1**
## Description
After further investigation it seems that Chrome does not allow any events for disabled buttons, as well as disables the bubbling of events. It seems to me the solution here is to just change the behaviour of react-bootstrap button to not apply disabled attribute to buttons but to apply custom css. The necessary css classes seem to be already present in mapstore. Currently disabled button is set to ignore onClick events, all other events go unimpeded.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue
#5926 

**What is the current behavior?**

#5926

**What is the new behavior?**
Disabled buttons have tooltips present.

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Review by MV88 — CHANGES_REQUESTED** [review]
i think the pr is good, but my concern are related to the fact that something should be done in order to prevent the usage of Button from react-bootstra to avoid in future a similar problem with disabled button,

I've done a quick investigation and it is possible for import statements by adding
```
"no-restricted-imports": ["error", {
        "paths": [{
            "name": "react-bootstrap",
            "importNames": ["Button"],
            "message": "Please import 'Button' from '/misc/Button' instead."
        }]
    }],
```

can you also test the rule for requires?
```
"no-restricted-modules": ["error", {
        "paths": [{
            "name": "react-bootstrap",
            "patterns": ["Button"],
            "message": "Please import 'Button' from '/misc/Button' instead."
        }]
```
 
@offtherailz what do you think about the eslint rules?

**Comment by vlt1** [comment]
@MV88 no-restricted-imports rule works but no-restricated-modules doesn't and I'm afraid it won't. You can't restrict using const {someProp} = require('someLib'); you can only restrict what kinds of strings can be passed to require(). The solution could be restricting react-bootstrap altogether and converting all existing react-bootstrap requires to using import.

**Comment by MV88** [comment]
> @MV88 no-restricted-imports rule works but no-restricated-modules doesn't and I'm afraid it won't. You can't restrict using const {someProp} = require('someLib'); you can only restrict what kinds of strings can be passed to require(). The solution could be restricting react-bootstrap altogether and converting all existing react-bootstrap requires to using import.

no need to change all  requires to import atm. thank you for testing. i think we can merge it and pay attention for the future to not use the react-bootstrap button, and use ours.


Now, please just solve the conflicts and we can merge it

**Review by MV88 — APPROVED** [review]

**Comment by tdipisa** [comment]
@vlt1 there are some conflicts to solve here. @MV88 as soon as conflicts have been fixed I think we can merge.

**Comment by offtherailz** [comment]
@MV88 now we merged eslint rules, should we do the eslint rule with a separated PR? I think we can postpone this

**Comment by MV88** [comment]
> @MV88 now we merged eslint rules, should we do the eslint rule with a separated PR? I think we can postpone this

yes we can do that in a separate pr

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,233 @@
+{
```
Comment (MV88) [inline .eslintrc:1]: please remove this file

we will port the changes to new rule in a separate pr

**Review by MV88 — CHANGES_REQUESTED** [review]

**Comment by MV88** [comment]
@vlt1 please check why the travis build is failing

**Comment by MV88** [comment]
closed in favor of https://github.com/geosolutions-it/MapStore2/pull/6212

**Pull request closed without merging**